core-bundle/tests/Search/DocumentTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Search;
+
+use Contao\CoreBundle\Search\Document;
+use Nyholm\Psr7\Uri;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class DocumentTest extends TestCase
+{
+    /**
+     * @dataProvider documentProvider
+     */
+    public function testDocument(string $body, array $expectedJsonLds): void
+    {
+        $document = new Document(
+            new Uri('https://example.com'),
+            200,
+            ['Content-Type' => ['text/html']],
+            $body
+        );
+
+        $this->assertSame('https://example.com', (string) $document->getUri());
+        $this->assertSame(200, $document->getStatusCode());
+        $this->assertSame(['content-type' => ['text/html']], $document->getHeaders());
+        $this->assertSame($expectedJsonLds, $document->extractJsonLdScripts());
+    }
+
+    public function testFactory(): void
```

Comment:
This method name does not describe what the method does.

Reply:
Well, it tests the factory does what it's supposed to do. That's it, I have nothing more to add here.